Update sl-jdbc.md

rpourzand opened the pull request:
We're making some changes to our API (not released yet), so I'm preemptively adding info.

## What are you changing in this pull request and why?
<!---
Describe your changes and why you're making them. If linked to an open
issue or a pull request on dbt Core, then link to them here! 

To learn more about the writing conventions used in the dbt Labs docs, see the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md).
-->

## Checklist
<!--
Uncomment if you're publishing docs for a prerelease version of dbt (delete if not applicable):
- [ ] Add versioning components, as described in [Versioning Docs](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-entire-pages)
- [ ] Add a note to the prerelease version [Migration Guide](https://github.com/dbt-labs/docs.getdbt.com/tree/current/website/docs/docs/dbt-versions/core-upgrade)
-->
- [ ] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.
- [ ] For [docs versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#about-versioning), review how to [version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) and [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content).
- [ ] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Adding new pages (delete if not applicable):
- [ ] Add page to `website/sidebars.js`
- [ ] Provide a unique filename for the new page

Removing or renaming existing pages (delete if not applicable):
- [ ] Remove page from `website/sidebars.js`
- [ ] Add an entry `website/static/_redirects`
- [ ] Run link testing locally with `npm run build` to update the links that point to the deleted page

rpourzand committed:
Update sl-jdbc.md

We're making some changes to our API (not released yet), so I'm preemptively adding info.

mirnawong1 committed:
Merge branch 'current' into rpourzand-patch-api-updates

DevonFulcher reviewed (approved):
Looks good, based on Paul's demo. I'm not sure when these features go live though.

mirnawong1 committed:
Merge branch 'current' into rpourzand-patch-api-updates

rpourzand committed:
Update sl-jdbc.md

Adding an example from Paul

rpourzand committed:
Update sl-graphql.md

copying the same over for jdbc -- maybe we can figure out a way to put this somewhere so we don't have to duplicate

rpourzand committed:
Update sl-jdbc.md

timedimension is no longer optional

rpourzand committed:
Update sl-jdbc.md

fixing some updates

rpourzand committed:
Update sl-graphql.md

fixing typo

plypaul commented on website/docs/docs/dbt-cloud-apis/sl-graphql.md:
```diff
@@ -432,25 +432,83 @@ mutation {
+
+Invalid Filters would be:
+
+  `* "{{ TimeDimension('metric_time') }} > '2020-01-01'"` - metrics in the query
```
The asterisk for the bullet point should be outside of the quote.

rpourzand replied:
good find!

plypaul commented on website/docs/docs/dbt-cloud-apis/sl-jdbc.md:
```diff
@@ -254,11 +254,70 @@ Where filters in API allow for a filter list or string. We recommend using the f
 
-Note: If you prefer a more explicit path to create the `where` clause, you can optionally use the `TimeDimension` feature. This helps separate out categorical dimensions from time-related ones. The `TimeDimesion` input takes the time dimension name and also requires granularity, like this: `TimeDimension('metric_time', 'MONTH')`.
+
+Note: If you prefer a more explicit path to create the `where` clause, you can optionally use the `TimeDimension` feature. This helps separate out categorical dimensions from time-related ones. The `TimeDimension` input takes the time dimension name optionally requires granularity, like this: `TimeDimension('metric_time', 'month')`.
```
Maybe `explicit path` -> `strongly typed` since that phrase was used before?

plypaul reviewed (approved):
LGTM aside form comments

plypaul commented:
The ambiguous resolution stack has not been released yet, so do we need to wait for that to go out before merging this?

rpourzand committed:
Update sl-graphql.md

making paul's reco's

rpourzand committed:
Update sl-jdbc.md

paul's recos

mirnawong1 committed:
Merge branch 'current' into rpourzand-patch-api-updates

rpourzand committed:
Update sl-graphql.md

also adding a dimension only example to it

Jstein77 commented on website/docs/docs/dbt-cloud-apis/sl-jdbc.md:
```diff
@@ -254,11 +254,70 @@ Where filters in API allow for a filter list or string. We recommend using the f
+
+Note: If you prefer a more explicit path to create the `where` clause, you can optionally use the `TimeDimension` feature. This helps separate out categorical dimensions from time-related ones. The `TimeDimension` input takes the time dimension name optionally requires granularity, like this: `TimeDimension('metric_time', 'month')`.
+
+For both `TimeDimension()` and `Dimension()` objects, the grain is only required if in the WHERE filter if the aggregation time dimensions for the measures & metrics associated with the where filter have different grains. 
```
Grain is still required when querying a time dimension using `Dimension()`. For example you would still need to add `Dimensions('metric_time','day')`.

rpourzand committed:
Update sl-jdbc.md

updating per jordan's input

mirnawong1 committed:
Update sl-graphql.md

format yaml

mirnawong1 commented on website/docs/docs/dbt-cloud-apis/sl-graphql.md:
```diff
@@ -428,29 +430,91 @@ mutation {
 
+For both `TimeDimension()` and `Dimension()` objects, the grain is only required in the WHERE filter if the aggregation time dimensions for the measures & metrics associated with the where filter have different grains. 
+
+For example, consider this Semantic Model and Metric Configuration which contains two metrics that are aggregated across different time grains. This example shows a single Semantic Model, but the same goes for metrics across more than one Semantic Model.
```
```suggestion
For example, consider this Semantic model and Metric configuration, which contains two metrics that are aggregated across different time grains. This example shows a single semantic model, but the same goes for metrics across more than one semantic model.
```

mirnawong1 committed:
Update website/docs/docs/dbt-cloud-apis/sl-graphql.md

mirnawong1 commented on website/docs/docs/dbt-cloud-apis/sl-jdbc.md:
```diff
@@ -254,11 +254,70 @@ Where filters in API allow for a filter list or string. We recommend using the f
 Where Filters have a few objects that you can use:
 
-- `Dimension()` - Used for any categorical or time dimensions. If used for a time dimension, granularity is required -  `Dimension('metric_time').grain('week')` or `Dimension('customer__country')`
+- `Dimension()` - Used for any categorical or time dimensions. `Dimension('metric_time').grain('week')` or `Dimension('customer__country')`
```
```suggestion
- `Dimension()` &mdash; Used for any categorical or time dimensions. `Dimension('metric_time').grain('week')` or `Dimension('customer__country')`
```

mirnawong1 committed:
Update website/docs/docs/dbt-cloud-apis/sl-jdbc.md

mirnawong1 commented on website/docs/docs/dbt-cloud-apis/sl-jdbc.md:
```diff
@@ -254,11 +254,70 @@ Where filters in API allow for a filter list or string. We recommend using the f
-- `Dimension()` - Used for any categorical or time dimensions. If used for a time dimension, granularity is required -  `Dimension('metric_time').grain('week')` or `Dimension('customer__country')`
+- `Dimension()` &mdash; Used for any categorical or time dimensions. `Dimension('metric_time').grain('week')` or `Dimension('customer__country')`
+
+- `TimeDimension()` - Used as a more explicit definition for time dimensions, optionally takes in a granularity `TimeDimension('metric_time', 'month')`
```
```suggestion
- `TimeDimension()` &mdash;  Used as a more explicit definition for time dimensions, optionally takes in a granularity `TimeDimension('metric_time', 'month')`
```

mirnawong1 committed:
Update website/docs/docs/dbt-cloud-apis/sl-jdbc.md

mirnawong1 commented on website/docs/docs/dbt-cloud-apis/sl-jdbc.md:
```diff
@@ -254,11 +254,70 @@ Where filters in API allow for a filter list or string. We recommend using the f
+
+- `TimeDimension()` &mdash;  Used as a more explicit definition for time dimensions, optionally takes in a granularity `TimeDimension('metric_time', 'month')`
 
 - `Entity()` - Used for entities like primary and foreign keys - `Entity('order_id')`
```
```suggestion
- `Entity()` &mdash;  Used for entities like primary and foreign keys - `Entity('order_id')`
```

mirnawong1 committed:
Update website/docs/docs/dbt-cloud-apis/sl-jdbc.md

mirnawong1 commented on website/docs/docs/dbt-cloud-apis/sl-jdbc.md:
```diff
@@ -254,11 +254,70 @@ Where filters in API allow for a filter list or string. We recommend using the f
+- `Entity()` &mdash;  Used for entities like primary and foreign keys - `Entity('order_id')`
+
+
+For `TimeDimension()`, the grain is only required in the WHERE filter if the aggregation time dimensions for the measures & metrics associated with the where filter have different grains. 
```
```suggestion
For `TimeDimension()`, the grain is only required in the `WHERE` filter if the aggregation time dimensions for the measures and metrics associated with the where filter have different grains. 
```

mirnawong1 committed:
Update website/docs/docs/dbt-cloud-apis/sl-jdbc.md

mirnawong1 commented on website/docs/docs/dbt-cloud-apis/sl-jdbc.md:
```diff
@@ -254,11 +254,70 @@ Where filters in API allow for a filter list or string. We recommend using the f
+
+For `TimeDimension()`, the grain is only required in the `WHERE` filter if the aggregation time dimensions for the measures and metrics associated with the where filter have different grains. 
+
+For example, consider this Semantic Model and Metric Config which contains two metrics that are aggregated across different time grains. This example shows a single Semantic Model, but the same goes for metrics across more than one Semantic Model.
```
```suggestion
For example, consider this Semantic model and Metric config, which contains two metrics that are aggregated across different time grains. This example shows a single semantic model, but the same goes for metrics across more than one semantic model.
```

mirnawong1 committed:
Update website/docs/docs/dbt-cloud-apis/sl-jdbc.md

mirnawong1 committed:
Update sl-jdbc.md

Commit:
add toggles

matthewshaver commented on website/docs/docs/dbt-cloud-apis/sl-jdbc.md:
```diff
@@ -254,11 +252,62 @@ Where filters in API allow for a filter list or string. We recommend using the f
 Where Filters have a few objects that you can use:
 
-- `Dimension()` - Used for any categorical or time dimensions. If used for a time dimension, granularity is required -  `Dimension('metric_time').grain('week')` or `Dimension('customer__country')`
+- `Dimension()` &mdash; Used for any categorical or time dimensions. `Dimension('metric_time').grain('week')` or `Dimension('customer__country')`
```
```suggestion
- `Dimension()` &mdash; Used for any categorical or time dimensions. `Dimension('metric_time').grain('week')` or `Dimension('customer__country')`.
```

matthewshaver commented on website/docs/docs/dbt-cloud-apis/sl-jdbc.md:
```diff
@@ -254,11 +252,62 @@ Where filters in API allow for a filter list or string. We recommend using the f
-- `Dimension()` - Used for any categorical or time dimensions. If used for a time dimension, granularity is required -  `Dimension('metric_time').grain('week')` or `Dimension('customer__country')`
+- `Dimension()` &mdash; Used for any categorical or time dimensions. `Dimension('metric_time').grain('week')` or `Dimension('customer__country')`
+
+- `TimeDimension()` &mdash;  Used as a more explicit definition for time dimensions, optionally takes in a granularity `TimeDimension('metric_time', 'month')`
```
```suggestion
- `TimeDimension()` &mdash;  Used as a more explicit definition for time dimensions, optionally takes in a granularity `TimeDimension('metric_time', 'month')`.
```

matthewshaver commented on website/docs/docs/dbt-cloud-apis/sl-jdbc.md:
```diff
@@ -254,11 +252,62 @@ Where filters in API allow for a filter list or string. We recommend using the f
+
+- `TimeDimension()` &mdash;  Used as a more explicit definition for time dimensions, optionally takes in a granularity `TimeDimension('metric_time', 'month')`
+
+- `Entity()` &mdash;  Used for entities like primary and foreign keys - `Entity('order_id')`
```
```suggestion
- `Entity()` &mdash;  Used for entities like primary and foreign keys - `Entity('order_id')`.
```

matthewshaver committed:
Apply suggestions from code review

mirnawong1 committed:
Merge branch 'current' into rpourzand-patch-api-updates

Jstein77 reviewed (approved):
This looks right to me. I think we need to add some more examples of how to query multi-hop dimensions in the CLI - but I can add that in a different PR.

rpourzand committed:
Update sl-graphql.md

rpourzand committed:
Update sl-graphql.md

rpourzand commented:
@mirnawong1  this is ready to merge!

mirnawong1 committed:
Merge branch 'current' into rpourzand-patch-api-updates

mirnawong1 commented on website/docs/docs/dbt-cloud-apis/sl-jdbc.md:
````diff
@@ -326,8 +376,9 @@ semantic_layer.query(metrics=['food_order_amount', 'order_gross_profit'],
-If you are ordering by an object that's been operated on (e.g., change granularity), or you are using the full object notation, descending order must look like:
+```
+
+If you are ordering by an object that's been operated on (e.g., you changed the the granularity of the time dimension), or you are using the full object notation, descending order must look like:
````
```suggestion
If you are ordering by an object that's been operated on (for example, you changed the the granularity of the time dimension), or you are using the full object notation, descending order must look like:
```

mirnawong1 committed:
Update website/docs/docs/dbt-cloud-apis/sl-jdbc.md

mirnawong1 commented on website/docs/docs/dbt-cloud-apis/sl-jdbc.md:
```diff
@@ -266,11 +264,62 @@ Where filters in API allow for a filter list or string. We recommend using the f
+  * `"{{ TimeDimension('metric_time', 'year') }} > '2020-01-01'"`
 
-Note: If you prefer a more explicit path to create the `where` clause, you can optionally use the `TimeDimension` feature. This helps separate out categorical dimensions from time-related ones. The `TimeDimesion` input takes the time dimension name and also requires granularity, like this: `TimeDimension('metric_time', 'MONTH')`.
+Invalid Filters would be:
```
```suggestion
Invalid filters would be:
```

mirnawong1 committed:
Update website/docs/docs/dbt-cloud-apis/sl-jdbc.md

mirnawong1 commented on website/docs/docs/dbt-cloud-apis/sl-jdbc.md:
```diff
@@ -364,16 +415,26 @@ semantic_layer.query(metrics=['food_order_amount', 'order_gross_profit'],
 ## FAQs
 
-<FAQ path="Troubleshooting/sl-alpn-error" />
+<FAQ path="Troubleshooting/sl-alpn-error" /><br />
```
```suggestion
<FAQ path="Troubleshooting/sl-alpn-error" />
```

mirnawong1 committed:
Update website/docs/docs/dbt-cloud-apis/sl-jdbc.md

mirnawong1 commented on website/docs/docs/dbt-cloud-apis/sl-graphql.md:
````diff
@@ -443,29 +445,91 @@ mutation {
 }
 ```
 
+For both `TimeDimension()`, the grain is only required in the WHERE filter if the aggregation time dimensions for the measures & metrics associated with the where filter have different grains. 
````
```suggestion
For both `TimeDimension()`, the grain is only required in the WHERE filter if the aggregation time dimensions for the measures and metrics associated with the where filter have different grains. 
```

mirnawong1 committed:
Update website/docs/docs/dbt-cloud-apis/sl-graphql.md

mirnawong1 committed:
Update sl-graphql.md

mirnawong1 committed:
Update sl-jdbc.md

mirnawong1 commented:
double checked and looking great! merging 🚢 - thank you so much @rpourzand !
